ci/ditt/main.py

```diff
@@ -125,80 +130,96 @@ def generateHTMLStatus(_htmlData, _cacheChanged):
         htmlData = []
         cacheChanged = False
 
+        NBL_DUMMY_CACHE_CASE = not bool(Path(str(NBL_CI_REFERENCES_DIR.absolute()) + '/' + NBL_CI_LDS_CACHE_FILENAME).is_file())
+        generatedReferenceCache = str(NBL_PATHTRACER_EXE.parent.absolute()) + '/' + NBL_CI_LDS_CACHE_FILENAME
+        destinationReferenceCache = str(NBL_CI_REFERENCES_DIR.absolute()) + '/' + NBL_CI_LDS_CACHE_FILENAME
+
+        sceneDummyRender = '"../ci/dummy_4096spp_128depth.xml"'
+        executor = str(NBL_PATHTRACER_EXE.absolute()) + ' -SCENE=' + sceneDummyRender + ' -TERMINATE'
+        subprocess.run(executor, capture_output=True)
+        
+        # if we start the path tracer first time
+        if NBL_DUMMY_CACHE_CASE:
+            shutil.copyfile(generatedReferenceCache, destinationReferenceCache)
+        elif not filecmp.cmp(destinationReferenceCache, generatedReferenceCache):
+            # fail CI if the reference cache is different that current generated cache
+            cacheChanged = True
+            CI_PASS_STATUS = False
```

Comment:
commit the cache file to the `ci/data/references`

always regenerate cache and compare

Reply:
obviously after you turn the `ci` folder into a submodule